pkg/logutil/log.go

```diff
@@ -107,24 +132,36 @@ func InitLogger(cfg *Config) error {
 	// error itself.
 	lg = lg.WithOptions(zap.AddStacktrace(zap.DPanicLevel))
 
-	log.ReplaceGlobals(lg, _globalP)
+	log.ReplaceGlobals(lg, globalP)
+	return nil
+}
 
-	var level zapcore.Level
-	err = level.UnmarshalText([]byte(cfg.Level))
-	if err != nil {
-		return errors.Trace(err)
+// InitLoggerWithWriteSyncer initializes a zap logger with specified write syncer.
+// For easy unit test when we use zaptest.Buffer for the zapcore.WriteSyncer
+func InitLoggerWithWriteSyncer(cfg *Config, output, errOutput zapcore.WriteSyncer) error {
```

Comment:
Is this function used in test only?

Reply:
Currently Yes. But user can customize their `WriteSyncer`.

Reply:
Can we put it in `Config` or an option?

Reply:
Move to option in latest commit.